### DISCUSSION TITLE
Getting Started?

### DISCUSSION THREAD

**Discussion by Sharkgamedev** [AlamoEngine-Tools/PetroglyphTools#176] [Q&A] [answered]
It appears there's no documentation for getting started with these tools.  
How would I set up a dev env for these?  
How would I use them in a project? Is there a pre-packaged version of these? I see @gruenwaldlk has some published on NUGET.

**Comment by gruenwaldlk** [comment, marked as answer]
Hello @Sharkgamedev,  
first of all, sorry for the late reply.  
You're right, there is no getting started whatsoever which we've never really thought about, seen as there's about 3 somewhat active developers for the project who all know each other.

Technically this very repository was meant as a development environment for the libraries, to make it easier to abstract common concepts away from the specific file type implementations or data unit representations (e.g. the data structure PG's binary formats share for complex data).
 The libraries (similar to the older versions I already published on NuGet) themselves would then be published via NuGet and GitHub Packages - we're just not at that stage yet. Probably won't be for a while either.

----

**So, getting started is a bit two-fold:**

If you want to actively use functionality, for now my published NuGet libraries should get you most of the way for the few file types I've published libraries for. Changes between those and the newer implementations here are mostly architecture, not functionality, because I built most of those in a rush to get some development automation tools done. Anything I haven't published, I never finished - some of it I ported here in whatever state it was in.

If you want to contribute to this project in some way, or want to use it as a foundation for something you're working on then at the moment you'd have to check out the project (or fork and check out) and open the top level with an IDE of your choice which should work - because there is a root level project, but I'Ve only ever used Rider to work on it.

**Reply by Sharkgamedev** [reply]
``Technically` this very repository was meant as a development environment for the libraries, to make it easier to abstract common concepts away from the specific file type implementations or data unit representations``  
Oh ok, that makes more sense. I was already using the old NuGet packages but wondered if this was a more up to date version of those ``Changes between those and the newer implementations here are mostly architecture,`` so good to hear this.  
  
``and open the top level with an IDE of your choice which should work - because there is a root level project, but I've only ever used Rider to work on it.`` Good to know - I suppose I'll find out if VS can handle it. 
 
Thanks for taking the time to answer this!
